tools/c7n_azure/c7n_azure/actions.py

```diff
@@ -411,6 +417,109 @@ def send_to_azure_queue(self, queue_uri, message):
         return StorageUtilities.put_queue_message(queue_service, queue_name, self.pack(message)).id
 
 
+DEFAULT_TAG = "custodian_status"
+
+
+class TagDelayedAction(BaseAction):
+    """Tag resources for future action.
+
+    The optional 'tz' parameter can be used to adjust the clock to align
+    with a given timezone. The default value is 'utc'.
+
+    If neither 'days' nor 'hours' is specified, Cloud Custodian will default
+    to marking the resource for action 4 days in the future.
+
+    .. code-block :: yaml
+
+      - policies:
+        - name: ec2-mark-for-stop-in-future
+          resource: azure.vm
+          filters:
+            - type: value
+              key: Name
+              value: instance-to-stop-in-four-days
+          actions:
+            - type: mark-for-op
+              op: stop
+    """
+
+    schema = utils.type_schema(
+        'mark-for-op',
+        tag={'type': 'string'},
+        msg={'type': 'string'},
+        days={'type': 'integer', 'minimum': 0, 'exclusiveMinimum': False},
+        hours={'type': 'integer', 'minimum': 0, 'exclusiveMinimum': False},
+        tz={'type': 'string'},
+        op={'type': 'string'})
+
+    default_template = 'Resource does not meet policy: {op}@{action_date}'
+
+    def __init__(self, data=None, manager=None, log_dir=None):
+        super(TagDelayedAction, self).__init__(data, manager, log_dir)
+        self.session = utils.local_session(self.manager.session_factory)
```

Comment:
Does self.session get used anywhere? May not need this.

Reply:
I didn't have this, but on line 52, it wants the session to get the api_version, and it complained when I left it out

Reply:
ah ok, missed that